server/protocols/jmap/src/main/java/org/apache/james/jmap/DownloadServlet.java

```diff
@@ -100,15 +112,23 @@ private void respondAttachmentAccessToken(MailboxSession mailboxSession, Downloa
         }
     }
 
-    private boolean attachmentExists(MailboxSession mailboxSession, String blobId) throws MailboxException {
+    private Optional<Blob> getBlob(MailboxSession mailboxSession, String blobId) throws MailboxException {
         try {
-            blobManager.retrieve(BlobId.fromString(blobId), mailboxSession);
-            return true;
+            Collection<MessageId> ownerMessageIds = attachmentManager.getOwnerMessageId(AttachmentId.from(blobId), mailboxSession);
```

Comment:
Security SHOULD be enforced by the AttachmentManager.

The AttachmentManager should not return attachments I do not have the rights on.